### Setup work space
```
mkdir KIAT
cd KIAT/
mkdir raw_data reference New_STAR
```
### Gathering files
```
cd /Network/Servers/avalanche.plb.ucdavis.edu/Volumes/Mammoth/Users/ruijuanli/2016_summer/raw_data
### Using fastq files suggested by Ruijuan
cp 50bp-05-11-Final-8ul_1.fq 50bp-05-11-Final-8ul_2.fq raw_data/
cd /Network/Servers/avalanche.plb.ucdavis.edu/Volumes/Mammoth/Users/ruijuanli/Reference/B.napus
cp star_genome Brassica_napus_v4.1.chromosomes.fa Brassica_napus.annotation_v5.gff3 reference/
```

### Trimming fq files from 101bp to 50bp
```
cd raw_data
fastx_trimmer -l 50 -i 05-11-Final-8ul_1.fq -o 50bp-05-11-Final-8ul_1.fq
fastx_trimmer -l 50 -i 05-11-Final-8ul_2.fq -o 50bp-05-11-Final-8ul_2.fq
cd ..
```

###Convert gff3 file to gtf to be used with STAR
```
cd reference/
gffread Brassica_napus.annotation_v5.gff3 -T -o Brassica_napus.annotation_v5.gtf
cd ..
```

###Running Paired End STAR on 50bp and 101bp reads
####Star Version 2.5.2b on Whitney, not using Coloma since it's on version 2.4.1d
```
cd New_Star/
mkdir 50bp.paired.star.dir 101bp.paired.star.dir
cd 50bp.paired.star.dir/
STAR --genomeDir ../../reference/star_genome --sjdbGTFfile ../../reference/Brassica_napus.annotation_v5.gtf --readFilesIn ../../raw_data/50bp-05-11-Final-8ul_1.fq ../../raw_data/50bp-05-11-Final-8ul_2.fq --outSAMtype BAM Unsorted SortedByCoordinate --outSAMunmapped Within --quantMode GeneCounts --twopassMode Basic --runThreadN 6
### Running next alignment on screen -r 16240.pts-40.whitney
cd KIAR/New_Star/101bp.paired.star.dir
STAR --genomeDir ../../reference/star_genome --sjdbGTFfile ../../reference/Brassica_napus.annotation_v5.gtf --readFilesIn ../../raw_data/05-11-Final-8ul_1.fq ../../raw_data/05-11-Final-8ul_2.fq --outSAMtype BAM Unsorted SortedByCoordinate --outSAMunmapped Within --quantMode GeneCounts --twopassMode Basic --runThreadN 6
```

###Troubleshooting poor GeneCounts file
The 50bp run above produced a count file which had zero counts for any gene.
Believe the created gtf file may be source of problem
Rerunning with gff3 file and new sjdb and quantMode parameters
Using screen -r 28681.pts-67.whitney
```
mkdir 50bp.paired.star.dir.2
STAR --genomeDir ../../reference/star_genome --sjdbGTFfile ../../reference/Brassica_napus.annotation_v5.gff3 --sjdbGTFtagExonParentTranscript Parent --sjdbGTFfeatureExon CDS --sjdbOverhang 49 --readFilesIn ../../raw_data/50bp-05-11-Final-8ul_1.fq ../../raw_data/50bp-05-11-Final-8ul_2.fq --outSAMtype BAM Unsorted SortedByCoordinate --outSAMunmapped Within --quantMode TranscriptomeSAM GeneCounts --twopassMode Basic --runThreadN 6
```
###Planned running cufflinks on output
```
cd 50bp.paired.star.dir/
mkdir cufflink_output      
cufflinks -u -p 16 -o cufflink_output --library-type fr-secondstrand -g ../../reference/Brassica_napus.annotation_v5.gtf -b ../../Brassica_napus_v4.1.chromosomes.fa Aligned.sortedByCoord.out.bam
cd ../101bp.paired.star.dir/
mkdir cufflink_output
cufflinks -u -p 16 -o cufflink_output --library-type fr-secondstrand -g ../../reference/Brassica_napus.annotation_v5.gtf -b ../../Brassica_napus_v4.1.chromosomes.fa Aligned.sortedByCoord.out.bam
```
