### Setup work space
```
mkdir KIAT
cd KIAT/
mkdir raw_data reference New_STAR Kallisto
```
### Gathering files
```
cd /Network/Servers/avalanche.plb.ucdavis.edu/Volumes/Mammoth/Users/ruijuanli/2016_summer/raw_data
### Using fastq files suggested by Ruijuan
cp 50bp-05-11-Final-8ul_1.fq 50bp-05-11-Final-8ul_2.fq raw_data/
cd /Network/Servers/avalanche.plb.ucdavis.edu/Volumes/Mammoth/Users/ruijuanli/Reference/B.napus
cp star_genome Brassica_napus_v4.1.chromosomes.fa Brassica_napus.annotation_v5.gff3 reference/
```

### Trimming fq files from 100bp to 50bp
```
cd raw_data
fastx_trimmer -l 50 -i 05-11-Final-8ul_1.fq -o 50bp-05-11-Final-8ul_1.fq
fastx_trimmer -l 50 -i 05-11-Final-8ul_2.fq -o 50bp-05-11-Final-8ul_2.fq
cd ..
```

###Convert gff3 file to gtf to be used with STAR
```
cd reference/
gffread Brassica_napus.annotation_v5.gff3 -T -o Brassica_napus.annotation_v5.gtf
cd ..
```
###Quick gene count with Kallisto
```
cd Kallisto
mkdir paired_100bp	paired_50bp	single_100bp	single_50bp
kallisto quant --plaintext -i ../reference/Brassica_napus.annotation_v5.cds.19.kai -o paired_50bp/ ../raw_data/50bp-05-11-Final-8ul_1.fq ../raw_data/50bp-05-11-Final-8ul_2.fq
kallisto quant --plaintext -i ../reference/Brassica_napus.annotation_v5.cds.19.kai -o paired_100bp/ ../raw_data/05-11-Final-8ul_1.fq ../raw_data/05-11-Final-8ul_2.fq
kallisto quant --single --plaintext -s 50 -l 250 -i ../reference/Brassica_napus.annotation_v5.cds.19.kai -o single_50bp/ ../raw_data/50bp-05-11-Final-8ul_1.fq
kallisto quant --single --plaintext -s 50 -l 250-i ../reference/Brassica_napus.annotation_v5.cds.19.kai -o single_100bp/ ../raw_data/05-11-Final-8ul_1.fq
cd
cd bin/
./combine_counts.sh
```

###Running Paired End STAR on 50bp and 101bp reads
####Star Version 2.5.2b on Whitney, not using Coloma since it's on version 2.4.1d
```
cd New_Star/
mkdir 50bp.paired.star.dir 100bp.paired.star.dir
cd 50bp.paired.star.dir/
STAR --genomeDir ../../reference/star_genome --sjdbGTFfile ../../reference/Brassica_napus.annotation_v5.gtf --readFilesIn ../../raw_data/50bp-05-11-Final-8ul_1.fq ../../raw_data/50bp-05-11-Final-8ul_2.fq --outSAMtype BAM Unsorted SortedByCoordinate --outSAMunmapped Within --quantMode GeneCounts --twopassMode Basic --runThreadN 6
### Running next alignment on screen -r 16240.pts-40.whitney
cd KIAR/New_Star/100bp.paired.star.dir
STAR --genomeDir ../../reference/star_genome --sjdbGTFfile ../../reference/Brassica_napus.annotation_v5.gtf --readFilesIn ../../raw_data/05-11-Final-8ul_1.fq ../../raw_data/05-11-Final-8ul_2.fq --outSAMtype BAM Unsorted SortedByCoordinate --outSAMunmapped Within --quantMode GeneCounts --twopassMode Basic --runThreadN 6
```

###Troubleshooting poor GeneCounts file
The 50bp run above produced a count file which had zero counts for any gene.
Believe the created gtf file may be source of problem
Log.out file says gene_id parameter was not found for every annotation
Rerunning with gff3 file and new sjdb and quantMode parameters
Using screen -r 28681.pts-67.whitney
```
mkdir 50bp.paired.star.dir.2
STAR --genomeDir ../../reference/star_genome --sjdbGTFfile ../../reference/Brassica_napus.annotation_v5.gff3 --sjdbGTFtagExonParentTranscript Parent --sjdbGTFfeatureExon CDS --sjdbOverhang 49 --readFilesIn ../../raw_data/50bp-05-11-Final-8ul_1.fq ../../raw_data/50bp-05-11-Final-8ul_2.fq --outSAMtype BAM Unsorted SortedByCoordinate --outSAMunmapped Within --quantMode TranscriptomeSAM GeneCounts --twopassMode Basic --runThreadN 6
```

Still did not work. Using R to edit gff3 file then repeat gff3 to gtf file conversion now
```
library(rtracklayer)
gff <- import.gff3("Brassica_napus.annotation_v5.gff3")
gff
gff$gene_id <- ifelse(is.na(gff$ID),gff$Parent,gff$ID)
export(gff,"Brassica_napus.annotation_v5.gff3",format="gff3")
```
```
gffread Brassica_napus.annotation_v5.gff3 -T -o Brassica_napus.annotation_v5.gtf
```

Rerunning on Coloma because Whitney is down
```
cd 50bp.paired.star.dir/
STAR --genomeDir ../../reference/star_genome --sjdbGTFfile ../../Tests/test.Brassica_napus.annotation_v5.gtf --sjdbOverhang 49 --readFilesIn ../../raw_data/50bp-05-11-Final-8ul_1.fq ../../raw_data/50bp-05-11-Final-8ul_2.fq --outSAMtype BAM Unsorted SortedByCoordinate --outSAMunmapped Within --quantMode GeneCounts --twopassMode Basic --runThreadN 9
cd KIAR/New_Star/100bp.paired.star.dir
STAR --genomeDir ../../reference/star_genome --sjdbGTFfile ../../Tests/test.Brassica_napus.annotation_v5.gtf --sjdbOverhang 99 --readFilesIn ../../raw_data/05-11-Final-8ul_1.fq ../../raw_data/05-11-Final-8ul_2.fq --outSAMtype BAM Unsorted SortedByCoordinate --outSAMunmapped Within --quantMode GeneCounts --twopassMode Basic --runThreadN 9
```
GeneCount file was not produced in this process for some reason
Trying a new test with new gff3 file instead of gtf file

###Method that worked on Whitney but not Coloma
```
cd 100bp.paired.star.dir/
STAR --genomeDir ../../reference/star_genome --sjdbGTFfile ../../Tests/test.Brassica_napus.annotation_v5.gff3 --sjdbGTFtagExonParentTranscript Parent --sjdbGTFfeatureExon CDS --sjdbOverhang 99 --readFilesIn ../../raw_data/05-11-Final-8ul_1.fq ../../raw_data/05-11-Final-8ul_2.fq --outSAMtype BAM Unsorted SortedByCoordinate --outSAMunmapped Within --quantMode TranscriptomeSAM GeneCounts --twopassMode Basic --runThreadN 9
cd ../50bp.paired.star.dir/
STAR --genomeDir ../../reference/star_genome --sjdbGTFfile ../../Tests/test.Brassica_napus.annotation_v5.gff3 --sjdbGTFtagExonParentTranscript Parent --sjdbGTFfeatureExon CDS --sjdbOverhang 49 --readFilesIn ../../raw_data/50bp-05-11-Final-8ul_1.fq ../../raw_data/50bp-05-11-Final-8ul_2.fq --outSAMtype BAM Unsorted SortedByCoordinate --outSAMunmapped Within --quantMode TranscriptomeSAM GeneCounts --twopassMode Basic --runThreadN 9
```
On Coloma get segmentation fault 11. After reading up on fault, may have been caused due to running out of memory on Coloma. Worked on Whitney. 

###Combining star read counts into one file
```
awk '{print $1 "\t" $2}' ReadsPerGene.out.tab| sed 1,4d | sed '1 i\ID 'single_50bp'' > ../Read.counts/single_50bp.counts
awk '{print $1 "\t" $2}' ReadsPerGene.out.tab| sed 1,4d | sed '1 i\ID 'single_100bp'' > ../Read.counts/single_100bp.counts
awk '{print $1 "\t" $2}' ReadsPerGene.out.tab| sed 1,4d | sed '1 i\ID 'paired_50bp'' > ../Read.counts/paired_50bp.counts
awk '{print $1 "\t" $2}' ReadsPerGene.out.tab| sed 1,4d | sed '1 i\ID 'paired_100bp'' > ../Read.counts/paired_100bp.counts
paste *.counts | awk '{ for (i=1;i<=NF;i+=2) {$i=""} print $0}' | sed 's/ /\t/g' > reads
paste *.counts | awk '{print $1}' > ID
paste ID reads > star.read.count.tsv
```

###Planned running cufflinks on output
```
cd 50bp.paired.star.dir/
mkdir cufflink_output      
cufflinks -u -p 9 -o cufflink_output --library-type fr-secondstrand -g ../../Tests/test.Brassica_napus.annotation_v5.gff3 -b ../../reference/Brassica_napus_v4.1.chromosomes.fa Aligned.sortedByCoord.out.bam
cd ../101bp.paired.star.dir/
mkdir cufflink_output
cufflinks -u -p 16 -o cufflink_output --library-type fr-secondstrand -g ../../reference/Brassica_napus.annotation_v5.gtf -b ../../Brassica_napus_v4.1.chromosomes.fa Aligned.sortedByCoord.out.bam
```
